build.gradle

```diff
@@ -227,30 +233,51 @@ subprojects {
             java {
                 srcDirs += generatedDir
             }
-            output.classesDir = sourceSets.main.output.classesDir
         }
         main {
+            compileClasspath += files([sourceSets.generated.java.outputDir])
+            runtimeClasspath += files([sourceSets.generated.java.outputDir])
             java {
-                compileClasspath += generated.output
-                runtimeClasspath += generated.output
-            }
-            scoverage {
-                compileClasspath += generated.output
-                runtimeClasspath += generated.output
+                exclude '**/*.jaxb'
+                exclude '**/README'
             }
         }
         test {
-            java {
-                compileClasspath += generated.output
-                runtimeClasspath += generated.output
-            }
-            testScoverage {
-                compileClasspath += generated.output
-                runtimeClasspath += generated.output
-            }
+            compileClasspath += configurations.compileOnly
```

Comment:
i think this was the reason i was using `provided` instead of `compileOnly`

Reply:
Yep. I'm sure it was.